File: neighbour.cpp

#include <iostream>
#include <string>
using namespace std;

class Neighbour {
private:
    string name;
    int age;
    string hobby;
    string personality;

public:
    Neighbour(string n, int a, string h, string p) {
        name = n;
        age = a;
        hobby = h;
        personality = p;
    }

    void introduce() {
        cout << "My neighbour's name is " << name << "." << endl;
        cout << "He is " << age << " years old." << endl;
        cout << "His hobby is " << hobby << "." << endl;
        cout << "Personality: " << personality << endl;
    }

    void greet() {
        cout << name << " says: 'Hey! How are you doing today?'" << endl;
    }

    void playHobby() {
        cout << name << " is playing " << hobby << " right now." << endl;
    }

    void visit() {
        cout << name << " just came over to visit." << endl;
    }

    void describePersonality() {
        cout << name << " is known to be " << personality << "." << endl;
    }
};

int main() {
    Neighbour neighbour1("Joshua Kinuthia", 20, "PES", "friendly and funny");

    neighbour1.introduce();
    neighbour1.greet();
    neighbour1.playHobby();
    neighbour1.visit();
    neighbour1.describePersonality();

    return 0;
}

File: neighbour.py

class Neighbour:
    def __init__(self, name, age, hobby, personality):
        self.name = name
        self.age = age
        self.hobby = hobby
        self.personality = personality

    def introduce(self):
        print(f"My neighbour's name is {self.name}.")
        print(f"He is {self.age} years old.")
        print(f"His hobby is {self.hobby}.")
        print(f"Personality: {self.personality}")

    def greet(self):
        print(f"{self.name} says: 'Hey! How are you doing today?'")

    def play_hobby(self):
        print(f"{self.name} is playing {self.hobby} right now.")

    def visit(self):
        print(f"{self.name} just came over to visit.")

    def describe_personality(self):
        print(f"{self.name} is known to be {self.personality}.")

# Create the neighbour object
neighbour1 = Neighbour(
    name="Joshua Kinuthia",
    age=20,
    hobby="PES",
    personality="friendly and funny"
)

neighbour1.introduce()
neighbour1.greet()
neighbour1.play_hobby()
neighbour1.visit()
neighbour1.describe_personality()
